client/src/api/profile.js

```diff
@@ -25,5 +26,12 @@ export const updateAccount = (name, division) => {
     name,
     division: Number.parseInt(division)
   })
-    .then(resp => resp.data)
+    .then(resp => {
+      switch (resp.kind) {
+        case 'badRateLimit':
+          toasts.useToast().add(resp.message)
+          return null
+      }
+      return resp.data
```

Comment:
have the success case (returning data that should only be valid when the response type is what you expect) in the switch and validate type; throw error in default case?